atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/service/SplitKeyDelegatingTransactionService.java

```diff
@@ -37,7 +36,11 @@
  * on which timestamps are requested. This class preserves the {@link TransactionService} guarantees regardless of
  * which underlying service is contacted.
  *
- * The timestamp service will throw an exception if the timestamp-to-service-key function returns a key which is
+ * The timestampToServiceKey function is expected to handle all possible long arguments (including negative values).
+ * The function may return null; if it does, then for reads, values written at that timestamp are considered to be
+ * uncommitted. The transaction service will throw if a write is attempted at such a timestamp.
```

Comment:
Not sure I follow this comment - isn't this natural if I haven't yet committed my transaction?

Reply:
This comment was to explain what the null values from #3694 mean, actually - since we have to allow for transaction managers that don't have access to a timestamp service. In this case, they are allowed to read, but may encounter values where coordination has not been achieved on the relevant start timestamp. These haven't committed (since to commit we need to putUnlessExists into the transactions service, which will achieve coordination on the start timestamp if needed), so the read-only services are allowed to recognise them as uncommitted.

Reply:
Thanks - this was due to my misunderstanding of what this layer was doing (serving the version of the store, not the commit timestamp directly).